### ISSUE TITLE
[Bug][BYT] Failed to load DSP topology on byt nocodec platform.

### ISSUE THREAD

**Issue by Jiangxinx** [thesofproject/linux#1145]
***Describe the bug***
probe of sof-nocodec failed with error -22 on byt-nocodec platform.

***dmesg***
```
[   11.864559] sof-audio-acpi 80860F28:01: src SRC3.0: source rate 0 sink rate 48000
[   11.864563] sof-audio-acpi 80860F28:01:  config: periods snk 2 src 2 fmt 2
[   11.864570] sof-audio-acpi 80860F28:01: ipc tx: 0x30010000: GLB_TPLG_MSG: COMP_NEW
[   11.864662] sof-audio-acpi 80860F28:01: ipc tx succeeded: 0x30010000: GLB_TPLG_MSG: COMP_NEW
[   11.864671] sof-audio-acpi 80860F28:01: tplg: ready widget id 22 pipe 3 type 31 name : BUF3.0 stream none
[   11.864676] sof-audio-acpi 80860F28:01: buffer BUF3.0: size 12288 caps 0x61
[   11.864682] sof-audio-acpi 80860F28:01: ipc tx: 0x30200000: GLB_TPLG_MSG: BUFFER_NEW
[   11.865011] sof-audio-acpi 80860F28:01: ipc tx succeeded: 0x30200000: GLB_TPLG_MSG: BUFFER_NEW
[   11.865022] sof-audio-acpi 80860F28:01: tplg: ready widget id 23 pipe 3 type 31 name : BUF3.1 stream none
[   11.865026] sof-audio-acpi 80860F28:01: buffer BUF3.1: size 12288 caps 0x41
[   11.865032] sof-audio-acpi 80860F28:01: ipc tx: 0x30200000: GLB_TPLG_MSG: BUFFER_NEW
[   11.867573] sof-audio-acpi 80860F28:01: ipc tx succeeded: 0x30200000: GLB_TPLG_MSG: BUFFER_NEW
[   11.867587] sof-audio-acpi 80860F28:01: tplg: ready widget id 24 pipe 3 type 31 name : BUF3.2 stream none
[   11.867593] sof-audio-acpi 80860F28:01: buffer BUF3.2: size 12288 caps 0x41
[   11.867600] sof-audio-acpi 80860F28:01: ipc tx: 0x30200000: GLB_TPLG_MSG: BUFFER_NEW
[   11.868390] sof-audio-acpi 80860F28:01: error: ipc error for 0x30200000 size 20
[   11.868408] sof-audio-acpi 80860F28:01: error: buffer BUF3.2 load failed
[   11.868419] sof-audio-acpi 80860F28:01: error: DSP failed to add widget id 0 type 31 name : BUF3.2 stream none reply -12
[   11.868459] sof-audio-acpi 80860F28:01: ASoC: failed to load widget BUF3.2
[   11.868500] sof-audio-acpi 80860F28:01: error: tplg component load failed -12
[   11.868585] sof-audio-acpi 80860F28:01: error: failed to load DSP topology -22
[   11.868597] sof-audio-acpi 80860F28:01: ASoC: failed to probe component -22
[   11.868638] sof-nocodec sof-nocodec: ASoC: failed to instantiate card -22
[   11.868693] sof-nocodec: probe of sof-nocodec failed with error -22
[   11.868724] sof-audio-acpi 80860F28:01: created machine sof-nocodec
[   12.310510] sof-audio-acpi 80860F28:01: ipc rx: 0x90020000: GLB_TRACE_MSG
[   12.310531] sof-audio-acpi 80860F28:01: ipc rx done: 0x90020000: GLB_TRACE_MSG
[   12.421002] systemd-udevd (250) used greatest stack depth: 11720 bytes left
[   16.424880] systemd-journald[215]: File /var/log/journal/810813fff71a4647924f113b7ac7bf25/user-1000.journal corrupted or uncleanly shut down, renaming and replacing.
[   17.398731] igb 0000:03:00.0 enp3s0: igb: enp3s0 NIC Link is Up 100 Mbps Full Duplex, Flow Control: RX/TX
[   17.398963] IPv6: ADDRCONF(NETDEV_CHANGE): enp3s0: link becomes ready
```

***Environment***
Firmware: db138eba https://github.com/thesofproject/sof/commits/master
Kernel: 2a22b942 https://github.com/thesofproject/linux/commits/topic/sof-dev
Topology: sof-byt-nocodec.tplg

**Comment by Jiangxinx** [comment]
Add bisect result:
Caused by sof#1728.

**Comment by juimonen** [comment]
fix is here https://github.com/thesofproject/sof/pull/1734

**Comment by Jiangxinx** [comment]
Fixed by thesofproject/sof#1734 .
